### ISSUE TITLE
Function invocation failed: TypeError: r is not a function

### ISSUE THREAD

**Issue by TylerMcGinn** [netlify/netlify-lambda#196]
There seems to be an issue with resolution of various packages during webpack and has something to do with require and import statements.  This was a previous issue and seems as though it has been closed without a fix. I have tried a potential fix mentioned on the previous discussion but which may have fixed the require/import problem but then introduced a different function invocation error - "Function invocation failed: Error: Cannot find module 'https". I should also mention the imported package that is causing this error for me is "faunadb".

```
//attempted fix by overriding  webpack config as mentioned in previous issue post
const webpack = require('webpack')

module.exports = {
    mode: "development",
    plugins: [
      new webpack.DefinePlugin({ "global.GENTLY": false })
    ],
  }
```
Link to previous issue post
[https://github.com/netlify/netlify-lambda/issues/64](url)

**Comment by sw-yx** [comment]
sorry you’re experiencing this. you’re going to have to debug this yourself unless someone comes along go help as i have no bandwidth to do it and you havent provided a repro. please read the readme to turn off minification and get some more insight into your errors. otherwise its gonna be up to you to resolve your fauna and webpack errors. sorry i know u wish i had some magic solution for you, but really you can see the entire source of netlify lambda here and i dont know any more than you do

**Comment by TylerMcGinn** [comment]
I ended up finding a fix to the problem. After commenting out my imports the breaking change pointed specifically at "faunadb". I eventually found that others have had the same problem with fauna DB and have raised the same issue on their repository as can be seen here.
 
[https://github.com/fauna/faunadb-js/issues/182](url)

 The problem is to do with fauna DB's current issued package and it sound like they realize this as well and should be implementing a fix shortly. The work around others are using is to rollback "faunadb" to version "2.7.0". This alone did not work for me but by using a webpack override with the previous code the problem was fixed. again here's my webpack.config.js file.
```
const webpack = require('webpack');

module.exports = {
    plugins: [
      new webpack.DefinePlugin({ "global.GENTLY": false })
    ],
  }
```

**Comment by MickCoelho** [comment]
Thanks @TylerMcGinn , I was running into the exact same issue and downgrading `faunadb` worked for me.

**Comment by sw-yx** [comment]
alright. i’ll close this and let fauna people know

**Comment by danieltodonnell** [comment]
Dan from Fauna here!  Just as a heads-up, this should be resolved in version 2.9 of our drivers to be released next week.  We appreciate everyone's patience with this matter and we are working hard to solve it!

Thanks!

**Comment by danieltodonnell** [comment]
For anyone blocked by this issue, I am pleased to report that you should now be UNblocked as-of version `faunadb@2.9.2`

https://www.npmjs.com/package/faunadb/v/2.9.2

This has been rigorously tested have a smooth experience.  If you experience any issues, feel free to tell us in Fauna Slack Community channel.

We apologize for the delay and inconvenience.  Thank you for your patience.
